server/broadcast.go

```diff
@@ -908,7 +911,7 @@ func transcodeSegment(ctx context.Context, cxn *rtmpConnection, seg *stream.HLSS
 		res, err = SubmitSegment(ctx, sess.Clone(), seg, nonce, calcPerceptualHash, verified)
 		if err != nil || res == nil {
 			if isNonRetryableError(err) {
-				cxn.sessManager.completeSession(sess)
+				cxn.sessManager.completeSession(ctx, sess, false)
 				return nil, info, err
 			}
 			cxn.sessManager.suspendAndRemoveOrch(sess)
```

Comment:
I was considering whether we should send a EndSessionRequest if we encounter an error such that we're suspending and removing the O from selection. On one hand, if we encounter an error here the O might not be in a state where it can process the EndSessionRequest (i.e. if we hit a timeout because it is not responding to requests). On another hand, if there is a chance that the O can handle the EndSessionRequest then that would be "cleaner". But, then there is also the possibility that the EndSessionRequest hangs and we need to make sure to set a proper timeout there...

Just laying out some thinking above...I suspect that it is fine to not send an EndSessionRequest here for now.